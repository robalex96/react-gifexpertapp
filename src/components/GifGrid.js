import React, { Fragment } from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {    
    const {data:gifs, loading} = useFetchGifs(category);

    return (
        <Fragment>
            <h3 className='animate__animated animate__fadeIn'>{category}</h3>
            {loading && <p>Loading...</p>}
            <div className='card-grid'>
                {
                    gifs.map(gif =>
                        (
                        <GifGridItem 
                            key={gif.id}
                            {...gif}
                        />
                        )
                    )
                }
            </div>
        </Fragment>
    )
};
