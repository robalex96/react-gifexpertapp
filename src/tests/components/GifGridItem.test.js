import React from 'react';
import { GifGridItem} from '../../components/GifGridItem'
import { shallow } from 'enzyme';

const title = 'Titulo prueba';
const url = 'https://localhost/prueba.jpg';

const wrapper = shallow(<GifGridItem title={title} url={url} />);

describe('GifGridItem Component', () => {
    test('Render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Title in p tag', () => {
       const p = wrapper.find('p'); 
       expect(p.text().trim()).toBe(title);
    });

    test('Img should have url and alt', () => {
        const {src, alt} = wrapper.find('img').props();
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Div should have class animation fadeIn', () => {
        const {className} = wrapper.find('div').props();
        expect(className.includes('animate__fadeIn')).toBe(true);
    });    
});