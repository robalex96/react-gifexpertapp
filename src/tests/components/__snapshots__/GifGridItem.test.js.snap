// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Component Render correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo prueba"
    src="https://localhost/prueba.jpg"
  />
  <p>
    Titulo prueba
  </p>
</div>
`;
